tests/integration/api/controllers/users.spec.js

```diff
@@ -1605,6 +1608,116 @@ describe('Users API', () => {
     });
   });
 
+  describe('GET api/v2/users/{username}', () => {
+    const password = 'password1234!';
+    let facility;
+    let person;
+    let user;
+    let userProgramOfficer;
+
+    before(async () => {
+      facility = placeFactory.place().build({ type: 'district_hospital', reported_date: null });
+      person = utils.deepFreeze(
+        personFactory.build({ role: 'chw', parent: { _id: facility._id }, reported_date: null })
+      );
+      facility.contact = person._id;
+      facility = utils.deepFreeze(facility);
+      user = userFactory.build({ username: 'chw111', place: facility._id, contact: person._id, roles: ['chw'] });
+      delete user.password;
+      user = utils.deepFreeze(user);
+      userProgramOfficer = utils.deepFreeze(userFactory.build({
+        username: 'prog222',
+        place: facility._id,
+        contact: person._id,
+        roles: ['program_officer']
+      }));
+
+      await utils.saveDocs([ facility, person ]);
+      await utils.createUsers([{ ...user, password }, { ...userProgramOfficer, password }]);
+
+      const settings = await utils.getSettings();
+      const permissions = {
+        ...settings.permissions,
+        'can_view_users': ['program_officer'],
+      };
+      await utils.updateSettings({ permissions }, true);
+    });
+
+    after(async () => {
+      await utils.revertSettings(true);
+      await utils.revertDb([], true);
+      await utils.deleteUsers([user]);
+    });
+
+    it('retrieves a user by username', async () => {
+      const users = await utils.request({
+        path: `/api/v2/users/${user.username}`,
+      });
+
+      expect(users).to.deep.include({
+        ...user,
+        place: { ...facility, _rev: users.place._rev },
+        contact: { ...person, _rev: users.contact._rev },
+      });
+    });
+
+    it('retrieves a user with can_view_users permission', async () => {
+      const users = await utils.request({
+        path: `/api/v2/users/${user.username}`,
+        auth: { username: userProgramOfficer.username, password },
+      });
+
+      expect(users).to.deep.include({
+        ...user,
+        place: { ...facility, _rev: users.place._rev },
+        contact: { ...person, _rev: users.contact._rev },
+      });
+    });
+
+    it('returns an error when no user is found for username', async () => {
+      try {
+        await utils.request({
+          path: `/api/v2/users/invalidUsername`,
+        });
+      } catch ({ error }) {
+        expect(error.code).to.equal(404);
+        expect(error.error).to.equal('Failed to find user with name [invalidUsername] in the [users] database.');
+        return;
+      }
+
+      expect.fail('Should have thrown an error');
```

Comment:
If you move this to the try block, after the `await utils.request`, you don't have to `return` in your catch.

Reply:
So, what I don't like about doing it that way is that the error generated by the `fail` gets caught in the `catch` block.  Then, if `request` call does succeed and the expected error is not thrown, the test will fail, but it will fail with an error like "Expected: Failed to find user... but got Should have thrown an error".  Not a huge thing, but can be a bit confusing when you go to start debugging.

This approach with the `return` is the most convenient I could think of, but open to other ideas!

Reply:
It's probably going to throw an error for `cannot read properties of undefined (reading 'code')` since the error being caught is destructured
The early return doesn't throw me off that much :sweat_smile:

Reply:
I logically prefer `expect.fail`-ing withing the try block. when I read this code, I immediately know: oh, this request is supposed to fail, and I prepare my brain for "why is it supposed to fail" and "how does the error look like". 

If I read the code the way it is now, top down, I see: oh, I'm OK with this request request not failing, but if it fails I run some assertions. Oh, I am not fine with this request failing.
If I read bottom up, I see an `expect.fail` at the end of the test and I think: oh, this is weird, so this test is _not_ supposed to run until the end. Now I have to search for all the returns and throws within the test. 

I totally understand that when reading error messages when the test fails, you get a prettier error message. My comment is about actually reading and understanding the test code. 

This is pretty minor and I'm not having a super strong opinion about this, it's just a preference about what makes sense and is easiest to follow for me when I read code.